src/features/validation/invalidDataValidation/useInvalidDataValidation.ts

```diff
@@ -0,0 +1,35 @@
+import { useMemo } from 'react';
+
+import dot from 'dot-object';
+
+import { FD } from 'src/features/formData/FormDataWrite';
+import { FrontendValidationSource, ValidationMask } from 'src/features/validation';
+import type { FieldValidations } from 'src/features/validation';
+
+const __default__ = {};
+
+export function useInvalidDataValidation(): FieldValidations {
+  const invalidData = FD.useInvalidDebounced();
+
+  return useMemo(() => {
+    if (Object.keys(invalidData).length === 0) {
+      return __default__;
+    }
+
+    return Object.keys(dot.dot(invalidData)).reduce((validations, field) => {
```

Comment:
Hmm, I would think it's possible for the invalidData object to a "shadow" of the full data model after a while, with empty objects, etc. It might be necessary to clean up/ignore such values before adding validation messages for everything in `dot.dot(invalidData)`?

Reply:
Yeah, I'm not 100% sure either. If you look at the result of `dot.dot(invalidData)` and it always only contains stringy values with invalid data (and not `null`, `{}`, etc), this isn't a problem after all. In older times we had the `flattenData()` function that did a `dot.dot()`, but also removed some values (such as empty objects), but it's worth noting that `dot.dot()` and that old `flattenData()` does not produce the same result (I've made that blunder a few times).

Reply:
Looks like empty objects will stay behind when removing nested properties, so I need some filtering I guess